src/groups/mqb/mqbstat/mqbstat_queuestats.cpp

```diff
@@ -512,101 +414,6 @@ QueueStatsDomain& QueueStatsDomain::setWriterCount(int writerCount)
-                                      value);
-    } break;
-    case EventType::e_PUSH: {
-        d_statContext_mp->adjustValue(DomainQueueStats::e_STAT_PUSH, value);
-    } break;
-    case EventType::e_PUT: {
-        d_statContext_mp->adjustValue(DomainQueueStats::e_STAT_PUT, value);
-    } break;
-    case EventType::e_ADD_MESSAGE: {
-        d_statContext_mp->adjustValue(DomainQueueStats::e_STAT_BYTES, value);
-        d_statContext_mp->adjustValue(DomainQueueStats::e_STAT_MESSAGES, 1);
-        if (!d_subContextsHolder.empty()) {
-            bsl::list<StatSubContextMp>::iterator it =
-                d_subContextsHolder.begin();
-            while (it != d_subContextsHolder.end()) {
-                it->get()->adjustValue(DomainQueueStats::e_STAT_BYTES, value);
-                it->get()->adjustValue(DomainQueueStats::e_STAT_MESSAGES, 1);
-                ++it;
-            }
-        }
-    } break;
-    case EventType::e_DEL_MESSAGE: {
-        d_statContext_mp->adjustValue(DomainQueueStats::e_STAT_BYTES, -value);
-        d_statContext_mp->adjustValue(DomainQueueStats::e_STAT_MESSAGES, -1);
-    } break;
-    case EventType::e_GC_MESSAGE: {
-        d_statContext_mp->adjustValue(DomainQueueStats::e_STAT_GC_MSGS, value);
-    } break;
-    case EventType::e_PURGE: {
-        // NOTE: Setting the value like that will cause weird results if using
-        //       the stat to get rates
-        d_statContext_mp->setValue(DomainQueueStats::e_STAT_BYTES, 0);
-        d_statContext_mp->setValue(DomainQueueStats::e_STAT_MESSAGES, 0);
-        if (!d_subContextsHolder.empty()) {
-            bsl::list<StatSubContextMp>::iterator it =
-                d_subContextsHolder.begin();
-            while (it != d_subContextsHolder.end()) {
-                it->get()->setValue(DomainQueueStats::e_STAT_BYTES, 0);
-                it->get()->setValue(DomainQueueStats::e_STAT_MESSAGES, 0);
-                ++it;
-            }
-        }
-    } break;
-    case EventType::e_CHANGE_ROLE: {
-        d_statContext_mp->setValue(DomainQueueStats::e_STAT_ROLE, value);
-    } break;
-    case EventType::e_CFG_MSGS: {
-        d_statContext_mp->setValue(DomainQueueStats::e_CFG_MSGS, value);
-    } break;
-    case EventType::e_CFG_BYTES: {
-        d_statContext_mp->setValue(DomainQueueStats::e_CFG_BYTES, value);
-    } break;
-    case EventType::e_NO_SC_MESSAGE: {
-        d_statContext_mp->adjustValue(DomainQueueStats::e_STAT_NO_SC_MSGS,
-                                      value);
-    } break;
-    case EventType::e_UPDATE_HISTORY: {
-        d_statContext_mp->setValue(DomainQueueStats::e_STAT_HISTORY, value);
-    } break;
-    default: {
-        BSLS_ASSERT_SAFE(false && "Unknown event type");
-    } break;
-    };
-}
-
 void QueueStatsDomain::onEvent(EventType::Enum    type,
                                bsls::Types::Int64 value,
                                const bsl::string& appId)
```

Comment:
This `onEvent` (per-appId one) is not templated/inlined in this PR. It's okay, we can do it later.